source/ship/ShipAICache.cpp

```diff
@@ -0,0 +1,129 @@
+/* ShipAICache.cpp
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ShipAICache.h"
+
+#include "../pi.h"
+#include "../Outfit.h"
+#include "../Ship.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+
+
+ShipAICache::ShipAICache(const Ship &ship)
+{
+	this->ship = &ship;
```

Comment:
Why are we passing a reference to the ship, just to turn it back into a pointer?

Reply:
If I contain a pointer I cant have a default constructor
Here the idea is to force to not be given nullptr, though ig it cant do that much harm